debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SourceInfo.java

```diff
@@ -48,13 +50,24 @@ public Lsn getCommitLsn() {
         return commitLsn;
     }
 
+    public Long getEventSerialNo() {
+        return eventSerialNo;
+    }
+
     /**
      * @param commitLsn - LSN of the {@code COMMIT} of the transaction whose part the change is
      */
     public void setCommitLsn(Lsn commitLsn) {
         this.commitLsn = commitLsn;
     }
 
+    /**
+     * @param eventSerialNo - order of the change with the same value of commit and change LSN
```

Comment:
Minor nit: there should be no "-", it will only mess up rendered JavaDoc.

Reply:
Have you checked both (all three messages)? The first one should have `1` and second/third should have `2`.

Reply:
This wasn't a PK update, so only one message is there.

Reply:
@gunnarmorling Strange as it may sound it is correct - and I've added explanation to the docs. The event counter is derived from events coming to the connector.
In case of PK update there are two events (co - commit id, ch - change id) - https://docs.microsoft.com/en-us/sql/relational-databases/system-tables/cdc-capture-instance-ct-transact-sql?view=sql-server-2017
```
co1:ch1 delete old PK - operation $1 - emitted with order id 1
co1:ch1 insert new PK - $operation $2 - emitted with order id 2
```
in case of regular update two evets will arrive
```
co2:ch2 update (old values) - operation $3 - merged with the next and filtered out
co2:ch2 update (new values) - $operation $4 - emitted with order id 2
```
So the order id does not related to output but to input sequence